packages/vitest/src/node/plugins/index.ts

```diff
@@ -56,7 +56,7 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest())
           options,
         )
         options.api = resolveApiConfig(options)
-        options.watch = options.watch && !options.run
+        options.watch = !options.run
```

Comment:
Should keep the original logic.

There are some logic under `dev` get removed in #702, we need to bring them back

Reply:
The logic of the `dev` function that got removed in #702 isn't needed anymore.
It was just acting as a default value, which is now included in `configDefaults`.